src/mod_muc_db.erl

```diff
@@ -0,0 +1,46 @@
+-module(mod_muc_db).
+-include("mod_muc.hrl").
+
+%% Defines which ODBC pool to use
+%% Parent host of the MUC service
+-type server_host() :: ejabberd:server().
+
+%% Host of MUC service
+-type muc_host() :: ejabberd:server().
+
+%% User's JID. Can be on another domain accessable over FED.
+%% Only bare part (user@host) is important.
+-type client_jid() :: ejabberd:jid().
+
+-type room_opts() :: [{OptionName :: atom(), OptionValue :: term()}].
+
+
+
+%% Called when MUC service starts or restarts for each domain
+-callback init(server_host(), ModuleOpts :: list()) -> ok.
+
+-callback store_room(server_host(), muc_host(), mod_muc:room(), room_opts()) ->
+    {aborted, _} | {atomic, _}.
+
+-callback restore_room(server_host(), muc_host(), mod_muc:room()) ->
+    {ok, room_opts()} | {error, room_not_found} | {error, term()}.
+
+-callback forget_room(server_host(), muc_host(), mod_muc:room()) -> ok.
+
+-callback can_use_nick(server_host(), muc_host(),
+                       client_jid(), mod_muc:nick()) -> boolean().
+
+-callback get_rooms(server_host(), muc_host()) -> [#muc_room{}].
```

Comment:
Again, I think we should standardize callback behaviour. This one returns a value or exception while other ones return `{ok, _}`, `{error, _}`.